samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/api.ts

```diff
@@ -0,0 +1,3318 @@
+      const localVarQueryParameter = {} as any;
+
+      // authentication api_key required
+      await setApiKeyToObject(
+        localVarHeaderParameter,
+        "api_key",
+        configuration
+      );
+
+      setSearchParams(localVarUrlObj, localVarQueryParameter);
+      let headersFromBaseOptions =
+        baseOptions && baseOptions.headers ? baseOptions.headers : {};
+      localVarRequestOptions.headers = {
+        ...localVarHeaderParameter,
+        ...headersFromBaseOptions,
+        ...options.headers,
+      };
+
+      return {
+        url: toPathString(localVarUrlObj),
+        options: localVarRequestOptions,
+      };
+    },
+    /**
+     *
+     * @summary Update an existing pet
+     * @param {Pet} body Pet object that needs to be added to the store
+     * @param {*} [options] Override http request option.
+     * @throws {RequiredError}
+     */
+    updatePet: async (
+      body: Pet,
+      options: RawAxiosRequestConfig = {}
+    ): Promise<RequestArgs> => {
+      // verify required parameter 'body' is not null or undefined
+      assertParamExists("updatePet", "body", body);
+      const localVarPath = `/pet`;
+      // use dummy base URL string because the URL constructor only accepts absolute URLs.
+      const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
+      let baseOptions;
+      if (configuration) {
+        baseOptions = configuration.baseOptions;
+      }
+
+      const localVarRequestOptions = {
+        method: "PUT",
+        ...baseOptions,
+        ...options,
+      };
+      const localVarHeaderParameter = {} as any;
+      const localVarQueryParameter = {} as any;
+
+      // authentication petstore_auth required
+      // oauth required
+      await setOAuthToObject(
+        localVarHeaderParameter,
+        "petstore_auth",
+        ["write:pets", "read:pets"],
+        configuration
+      );
+
+      localVarHeaderParameter["Content-Type"] = "application/json";
+
+      setSearchParams(localVarUrlObj, localVarQueryParameter);
+      let headersFromBaseOptions =
+        baseOptions && baseOptions.headers ? baseOptions.headers : {};
+      localVarRequestOptions.headers = {
+        ...localVarHeaderParameter,
+        ...headersFromBaseOptions,
+        ...options.headers,
+      };
+      localVarRequestOptions.data = serializeDataIfNeeded(
+        body,
+        localVarRequestOptions,
+        configuration
+      );
+
+      return {
+        url: toPathString(localVarUrlObj),
+        options: localVarRequestOptions,
+      };
+    },
+    /**
+     *
+     * @summary Updates a pet in the store with form data
+     * @param {number} petId ID of pet that needs to be updated
+     * @param {string} [name] Updated name of the pet
+     * @param {string} [status] Updated status of the pet
+     * @param {*} [options] Override http request option.
+     * @throws {RequiredError}
+     */
+    updatePetWithForm: async (
+      petId: number,
```

Comment:
shouldnt these be combined into a single request parameters? are the samples up-to-date?

Reply:
They likely should be. My change was focused on the `*Interface` component that starts on line [1180](https://github.com/OpenAPITools/openapi-generator/pull/19467/files#diff-8bf02f07382b9f8b689c969baf8181bc2d9116c07d108b6405b832185427bb91R1180), but you are right the other components are not updated.

I'll look into updating those as well!

Reply:
@macjohnny I looked into this a bit and realized to update the ParamCreator and Fp style would be a pretty major change. The main benefit of this change is it aligns the interface to the default API client, which I believe is what most consumers of a generate client would use anyway. 

Just bringing this up as I think there might be some tradeoffs to consider between solving the immediate bug that opened the initial issue (https://github.com/OpenAPITools/openapi-generator/issues/6783) or trying to reset all of the other components in the template

Reply:
ok makes sense, lets fix this step by step